react/features/filmstrip/components/web/Thumbnail.tsx

```diff
@@ -1169,20 +1164,10 @@ function _mapStateToProps(state: IReduxState, ownProps: any): Object {
     const participant = getParticipantByIdOrUndefined(state, participantID);
     const id = participant?.id ?? '';
     const isLocal = participant?.local ?? true;
-    const multipleVideoSupportEnabled = getMultipleVideoSupportFeatureFlag(state);
-    const _isVirtualScreenshareParticipant = multipleVideoSupportEnabled && isScreenShareParticipant(participant);
+    const _isVirtualScreenshareParticipant = isScreenShareParticipant(participant);
     const tracks = state['features/base/tracks'];
-
-    let _videoTrack;
-
-    if (_isVirtualScreenshareParticipant) {
-        _videoTrack = getVirtualScreenshareParticipantTrack(tracks, id);
-    } else {
-        _videoTrack = isLocal
-            ? getLocalVideoTrack(tracks) : getTrackByMediaTypeAndParticipant(tracks, MEDIA_TYPE.VIDEO, participantID);
-    }
-    const _audioTrack = isLocal
-        ? getLocalAudioTrack(tracks) : getTrackByMediaTypeAndParticipant(tracks, MEDIA_TYPE.AUDIO, participantID);
+    const _videoTrack = getVideoTrackByParticipant(state, participant);
```

Comment:
Why do we skip the check for local here? Before if the participant have been undefined we would select the local track and now the tracks are going to be undefined.

Reply:
When will participant be undefined ? All the local tracks will also have participant id and we should be able to do a look up based on just the mediaType and participantID right ?

Reply:
So as I'm looking into the code the flow is the following:
 - We create a local participant without id.
 - we set the local participant id of 'local'
 - we join the conference and we set the resource of the jid as an id.

So we definitely have both - a time when we don't have local participant, we have a local participant with a dummy id.

Now `getLocalVideoTrack` and `getLocalAudioTrack` are filtering the tracks only by the `.local` property.

I remember doing some fixes a long time ago here on the thumbnails around this. That's why I'm pointing it out.

I guess the video use case will probably work because maybe the tracks will have participantID 'local'. Is this the case? 

And what will happen for the period of time when we don't have a local participant? Or we create it so early that React haven't rendered anything yet? Have you checked this?

For the audio use case (the next line) I think we should definitely take extra care because `participantID` prop can definitely be set to `undefined`. I think we were doing that for the local participant always. If this is not the case anymore maybe we can remove the checks for handling `participantID === undefined`.